### ISSUE TITLE
`Extraction TestCompile` should clean up after itself, even when it fails

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#16683]
#### Description of the problem

I left a bug minimization job running overnight, and I ran out of hard disk space because `Extraction TestCompile` called repeatedly resulted in a 32 GB `/tmp` folder.

#### Coq Version

8.15

**Comment by herbelin** [comment]
The motivation for keeping the tmp file in case of failure was to be able to analyze it. What you are suggesting is that it conflicts with other needs.

**Comment by JasonGross** [comment]
Yes.  Maybe make a flag (off by default, ideally) for keeping temporaries?  (Off by default because otherwise the minimizer will need fairly complicated logic to prevent itself from removing whatever turns the flag off, if it's on by default.)

**Comment by SkySkimmer** [comment]
Native compile uses a debug flag that way.

**Comment by SkySkimmer** [comment]
Also maybe the minimizer could handle some tmp cleanup itself? something like 
~~~bash
OLD_TMPDIR=$TMPDIR
export TMPDIR=$(mktemp -d)
coqc ...
rm -rf $TMPDIR
TMPDIR=$OLD_TMPDIR
~~~

**Comment by JasonGross** [comment]
Ah, `coqc` respects `TMPDIR`?
